### ISSUE TITLE
Allow currentColor

### ISSUE THREAD

**Issue by sbaechler** [sh-waqar/stylelint-declaration-use-variable#14]
Allow the magic color value `currentColor` which stands for the current text color.

**Comment by okonet** [comment]
Do mean the plugin should warn about such usages and enforce a variable? Since `currentColor` is inferred from the CSS cascade and there is no way of calculating it statically.

**Comment by sbaechler** [comment]
No, `currentColor` is good. I would allow it even if `use-variable` was enforced.

**Comment by AndyOGo** [comment]
I would argue to allow for a whitelist, for stuff like `inherit`, `currentColor`, `transparent` etc.

**Comment by AndyOGo** [comment]
@sbaechler @okonet @sh-waqar 
I suggest the following optional syntax for exceptions being of type `string`, `function`, `RegEx`:

````json
{
  "sh-waqar/declaration-use-variable": [["/color/", "fill", "stroke", "z-index", "font-size"], {
    "except": ["inherit", "currentColor", "transparent"]
   }],
}
````

**Comment by sh-waqar** [comment]
@AndyOGo - Looks good, it will solve a lot of issues and will give a flexible api to developers. I am not sure if stylelint can provide it but it can be done programmatically. If you want you can fork the project and raise a PR and I would be happy to merge.

**Comment by AndyOGo** [comment]
@sbaechler @okonet @sh-waqar
Anyone who is interested I just released my own plugin which implements this feature.
Go and check it: https://github.com/AndyOGo/stylelint-declaration-strict-value
